### PR TITLE
Sbaud/iac diff scan

### PR DESCRIPTION
**Pull request by sylvain-baud-gg**
Add iac diff scan to pyGitGuardian.

We use the same implementation logic than iac directory scan.

Careful, just like iac directory scan, iac diff scan needs to access the data argument of request. Therefore iac_diff_scan bypass the GGClient.post method and directly call GGClient.request.

### CODE REVIEW

File: changelog.d/20230703_154225_sylvain.baud.ext_iac_diff_scan.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+### Added
+
+- iac_diff_scan added to GGClient. This allow to scan two directories and categorised found invidents as new, unchanged or deleted.
```
Comment (gg-teolohrer) [inline changelog.d/20230703_154225_sylvain.baud.ext_iac_diff_scan.md:3]:
```suggestion
- `iac_diff_scan` added to GGClient. This allows scanning two directories for IaC vulnerabilities and categorise incidents as new, unchanged or deleted.
```

**Review by GG-HH — APPROVED** [review]

**Review by agateau-gg — CHANGES_REQUESTED** [review]
Minor remark on doc.

File: pygitguardian/client.py [hunk 2]
```diff
@@ -498,6 +504,41 @@ def iac_directory_scan(
 
         return result
 
+    # For IaC diff Scans
```
[the remark this answers is not in the document]
Comment (sylvain-baud-gg) [reply pygitguardian/client.py]: Added docstring to `iac_directory_scan` and `iac_diff_scan`

**Review by agateau-gg — APPROVED** [review]
Nice! Thanks for adding a docstring to `iac_directory_scan()` too! :+1:

**Pull request merged**